test/types/mongodb.test-d.ts

```diff
@@ -23,13 +23,13 @@ expectNotDeprecated(MongoDBDriver.ObjectId);
 const client = new MongoClient('');
 const coll = client.db('test').collection('test');
 const findCursor = coll.find();
-expectType<Document | null>(await findCursor.next());
+expectType<WithId<Document> | null>(await findCursor.next());
 const mappedFind = findCursor.map<number>(obj => Object.keys(obj).length);
 expectType<FindCursor<number>>(mappedFind);
 expectType<number | null>(await mappedFind.next());
 expectType<number[]>(await mappedFind.toArray());
 const aggCursor = coll.aggregate();
-expectType<Document | null>(await aggCursor.next());
+expectType<WithId<Document> | null>(await aggCursor.next());
```

Comment:
We'll need to consider more testing. If you look in `test/types/community/collection/findX.test-d.ts` there's some example schemas making use of the find APIs. And `test/types/community/cursor.test-d.ts` has cursor API tests.